packages/ui/src/common/OTPField.tsx

```diff
@@ -0,0 +1,68 @@
+import * as Common from '@clerk/elements/common';
+import React from 'react';
+
+import * as Field from '../primitives/field';
+
+export function OTPField({
+  label = 'One-time password',
+  ...props
+}: React.ComponentProps<typeof Common.Input> & {
+  label?: React.ReactNode;
+}) {
+  return (
+    <Common.Field
+      asChild
+      name='code'
+    >
+      <Field.Root>
+        <Common.Label asChild>
+          <Field.Label visuallyHidden>{label}</Field.Label>
+        </Common.Label>
+
+        <Common.FieldState>
+          {({ state }) => (
+            <Common.Input
+              type='otp'
+              autoSubmit
+              className='flex gap-2'
+              render={({ value, status }) => (
+                <Field.Input
+                  type='otp'
+                  data-status={status}
+                  intent={state}
```

Comment:
Would this be `data-intent`? or do we need this since data-status is above?

Reply:
In this case we're passing the `intent` down to the `Field` which will handle the appropriate error/success/warning styles